proto/telemetry.proto

```diff
@@ -0,0 +1,104 @@
+syntax = "proto3";
+
+import "google/rpc/status.proto";
+
+package telemetry;
+
+// Request containing telemetry data to log.
+message LogTelemetryRequest {
+  // Telemetry logs to store. Can contain multiple logs if previous requests
+  // failed. If a log with matching UUIDs was previously uploaded, those logs
+  // will be silently ignored.
+  repeated TelemetryLog log = 1;
+}
+
+// Response from logging telemetry containing the status of the log and
+// information about whether or not a BuildBuddy upgrade is available.
+message LogTelemetryResponse {
+  // The status of the log request. If status code is OK (0), logs were
+  // successfully uploaded. Requests should be retried for non-zero statuses.
+  google.rpc.Status status = 3;
+
+  // Information about whether or not a BuildBuddy upgrade is available.
+  UpgradeInfo upgrade_info = 2;
+}
+
+// Represents a single sample of telemetry data.
+message TelemetryLog {
+  // A unique identifier representing the BuildBuddy installation. This
+  // identifier is persisted across runs and shared across instances.
+  string installation_uuid = 1;
+
+  // A unique identifier representing the BuildBuddy instance that made the
+  // request. This identifier is stored in memory and will not survive restarts.
+  string instance_uuid = 2;
+
+  // A unique identifier representing the log used for deduplication.
+  string log_uuid = 3;
+
+  // The client side timestamp in microseconds at which this log was recorded.
+  int64 recorded_at_usec = 4;
+
+  // The BuildBuddy version.
+  string app_version = 5;
+
+  // The URL at which BuildBuddy is running.
+  string app_url = 6;
+
+  // The hostname of this BuildBuddy instance.
+  string hostname = 7;
+
+  // Daily BuildBuddy usage statistics.
+  TelemetryStat daily_stat = 8;
```

Comment:
sure you don't want to just call the fields telemetry_stat and telemetry_feature?

Reply:
Done